src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -123,20 +126,26 @@ of this software and associated documentation files (the "Software"), to deal
      * @param clientID The client ID for the created OAuth Application.
      * @param clientSecret The client secret for the created GitHub OAuth Application.
      * @param oauthScopes A comma separated list of OAuth Scopes to request access to.
+     * @param emailDomains An optional comma separated list of domain(s) to select for email
+     * @param forceGithubEmail Force the email from github to override the one in the profile
      */
     @DataBoundConstructor
     public GithubSecurityRealm(String githubWebUri,
             String githubApiUri,
             String clientID,
             String clientSecret,
-            String oauthScopes) {
+            String oauthScopes,
+            String emailDomains,
+            Boolean forceGithubEmail) {
```

Comment:
This will not cleanly migrate configuration.  Please overload this method with another deprecated method so that configurations cleanly migrate.

A simple test:
1. Build this plugin.
2. Start Jenkins with the stable GitHub OAuth plugin configured.
3. Upgrade to this built plugin and see how the configuration migration behaves.

I'll take a closer look at this during the week.  There are other plugins which do as I describe.  Try to find examples there.  Otherwise, I'll try to find an example.